### ISSUE TITLE
gputemperature@silentage.com: only basic info shown with AMDGPU and only visible from top panel

### ISSUE THREAD

**Issue by some-username-here1** [linuxmint/cinnamon-spices-applets#2448]
* Applet version/Build date: `2.0.0`/`2019-03-20 02:32 UTC`
 * Cinnamon version: Cinnamon 4.0.10
 * Distribution: Linux Mint 19.1 "Tessa" Cinnamon (64-bit)
 * GPU: `XFX Radeon RX 580 GTR 8GB`
   * Drivers: AMDGPU (`19.0.1+git1905091930.4b1753~oibaf~b`)
 * Snippet or attachment of ~/.cinnamon/glass.log [file is non-existent] 
 * [xsession-errors.txt](https://github.com/linuxmint/cinnamon-spices-applets/files/3277334/xsession-errors.txt)

## Notify author of applet
**[N/A]** (author no longer exists on GitHub.)

## Issue
1. Limited to basic function (only showing the GPU temp, nothing else) on AMDGPU driver, and 
2. limited visibility to the top panel, unable to show from the bottom- or side-panels

## Steps to reproduce
Issue 1 (Basic functionality):
* having an AMD Graphics Card capable of full support for AMDGPU

Issue 2 (only visible from top panel issue):
1. Add two applets of `gputemperature`, one on a top panel, and second on a bottom panel.
2. Try to open both, only to see that the one on the top panel will be visible.

## Expected behaviour:
`gputemperature` looking like the middle-left example, pulled from the applet page on Cinnamon Spices:
![image](https://user-images.githubusercontent.com/34193156/59285011-1f18b380-8c22-11e9-91e7-bc523709c921.png)
Along with it being able to work from bottom or side taskbars (not shown)

## Other information
Cinnamon warning about the applet with "This applet contains function calls that could potentially cause Cinnamon to crash or freeze. If you are experiencing crashes or freezing, please try removing it."

**Comment by Odyseus** [comment]
Hello, @some-username-here1.

> Limited to basic function...

That's the expected output for your hardware. What you see in the screenshot from the Cinnamon Spices website for this applet is the output of an nVidia card, [which the applet gathers using the `nvidia-settings` command](https://github.com/linuxmint/cinnamon-spices-applets/blob/master/gputemperature%40silentage.com/files/gputemperature%40silentage.com/applet.js#L194). In your case, using an ATI card, the applet gathers only the cards' temperature [by parsing the output of the `sensors` command](https://github.com/linuxmint/cinnamon-spices-applets/blob/master/gputemperature%40silentage.com/files/gputemperature%40silentage.com/applet.js#L95).

> Limited visibility to the top panel...

The applet is not multi-instance. There can be only one instance of the applet placed on any panel. And even if it were set to be multi-instance, [it isn't initialized as such](https://github.com/linuxmint/cinnamon-spices-applets/blob/master/gputemperature%40silentage.com/files/gputemperature%40silentage.com/applet.js#L886).

> Cinnamon warning about the applet with...

That's just a warning. It is triggered by the applet using some function calls using synchronous operations (`GLib.spawn_command_line_sync`).

**Comment by some-username-here1** [comment]
Oh, thanks for the clarification. About the second thing, though, is that I don't mean it in having more than one instance of it, sorry for the confusion on that, but having it on the bottom taskbar instead of the top doesn't allow it to be shown when clicked on:
Top:
![image](https://user-images.githubusercontent.com/34193156/59318110-c45f7600-8c7a-11e9-9f0c-b8af887a56e0.png)
Bottom (I did try clicking on it, but nothing came up from it):
![image](https://user-images.githubusercontent.com/34193156/59318230-1e603b80-8c7b-11e9-9d53-53b77aef979a.png)
unless this was intentional? and I would fix this if I knew how to code, but I'm still an amateur on it

**Comment by Odyseus** [comment]
> unless this was intentional?

No, it isn't intentional. What happens is that the applet code hasn't been updated (its functional parts at least) in more than three years. More than 6 mayor Cinnamon versions have existed since then. And a lot of changes in core functionality.

Your problem could be that the applet doesn't handle orientation changes (moving the applet from the top to the bottom). Try restarting Cinnamon after you place the applet in the bottom panel. That might restore its functionality.

**Comment by Stoatwblr** [comment]
The applet doesn't work with AMDGPU here. All it displays is NaN

lmsensors says:
amdgpu-pci-0100
Adapter: PCI adapter
vddgfx:           N/A  
fan1:           0 RPM
edge:         +60.0°C  (crit = +120.0°C, hyst = +90.0°C)
ERROR: Can't get value of subfeature power1_cap: Can't read
power1:           N/A  (cap =   0.00 W)


Works fine with the radeon driver. 
(Ubuntu 19.10 with Cinnamon desktop loaded, 5 monitors, 1 workspace, applet shows the same error in all of them)
